src/graphql_clj/box.clj

```diff
@@ -0,0 +1,38 @@
+(ns graphql-clj.box
+  (:import [clojure.lang IObj IDeref IPersistentCollection Named]
+           [java.io Writer]))
+
+(deftype Box [value meta]
+
+  IObj
+  (meta [_] meta)
+
+  IDeref
+  (deref [_] (if (instance? Box value) @value value))
+
+  IPersistentCollection
+  (equiv [a b] (.equals a b))
+
+  Named
+  (getName [this] (if (instance? Box value) (.getName value) (.toString this)))
+
+  Object
+  (equals [_ b] (if (instance? Box b) (= value @b) (= value b)))
```

Comment:
This is here to make this work for test expectation purposes only - we aren't relying on equality anywhere in the validation code.  I tried to extend the type in test helpers only, but it seems that extend-type only works for protocols, not Java interfaces.  As long as this type doesn't leak out of the validation phase (we are removing it before validation completes), I think it's ok for now until we have a better / cleaner idea, as this is clearly suboptimal.